### ISSUE TITLE
Markdown live preview

### ISSUE THREAD

**Issue by bummytime** [Automattic/simplenote-macos#326]
Currently when markdown is enabled in the editor, the user has to flip to preview mode in order to see how their marked-up text renders:

![markdown_preview](https://user-images.githubusercontent.com/154014/57783461-c245de00-76f3-11e9-9b1f-2da90086ec21.gif)

This dual-mode experience presents a few issues:
* It's an extra step/click to get to preview mode
* Many users continually flip back and forth between modes while editing
* (Subjective) Markdown markup doesn't look as nice as already-formatted text
* Users coming from more traditional editors (Notes.app, Word, etc) are not used to this
* The markup causes the user to lose focus on their content

So, my suggestion here is that we implement a live markdown preview where markup is rendered in real time. At the lowest levels, notes are still markdown...stored exactly as they are today. There are a few editors that already do this — a great example is [Typora](https://typora.io/).

Additionally, this new mode could be a global (and/or note-level) setting (on by default) that users can disable if they prefer the "classic" Simplenote editor experience.

**Comment by zero-hugo** [comment]
This is exactly what I want to say.A live markdown preview will be very useful .

**Comment by ReessKennedy** [comment]
+1

**Comment by cstby** [comment]
Typora UX is not as different as it may seem. Typora has a "source" mode that shows all the markup.

simplenote could get closer by
- changing the preview mode to a live preview
- making the live preview the default for a markdown enabled files
- providing a keyboard shortcut to switch between the live preview and source

**Comment by rollecode** [comment]
As a long time fan of Typora I'd love to see this implemented in Simplenote.

**Comment by roorco** [comment]
+1

**Comment by Duseylicious** [comment]
This is the difference between me using it and not.  I love how simple it is, the quick hyperlinking to internal pages, etc.  But I want to see the preview while I'm editing, the vast majority of the time.

**Comment by david-nyt** [comment]
I agree with @Duseylicious, this is a key feature for me and without it I won't use Simplenote. Most of my notes are living documents I'm consistently going back to look at what I wrote and adding more, so it would be very annoying to keep switching back and forth between the preview mode. A live preview would make Simplenote perfect for me.

**Comment by rollecode** [comment]
Just to chime in, I have basically been using Simplenote as simple "raw text notepad" without any formatting. Some notes get really messy because of that.  I personally don't undestand _any_ markdown editor that has "code view" and "preview view" side by side. The power of markdown is simplicity with less formatting options than for example Google Docs, I can't comprehend why the combined editing has not been utilized more. Typora would be perfect but lacks combined notes view with real time saving to the cloud. Google Docs is too versatile for simple temp note taking and is more meant for official documents. Simplenote has everything except full markdown editing support with live preview. This should be considered more highly.

**Comment by rollecode** [comment]
Year has passed. Been using [marktext](https://github.com/marktext/marktext) and lately [Nota](https://nota.md/) for raw md editing because they have beautiful live markdown editors. Sadly they are _just editors_ for desktop so there's no cloud sync or mobile apps. This is why for quick notes Simplenote still is the best thing out there and I'll keep using it till the end of times.

Anyway, just wanted to bump with these two excellent app examples for inspiration.

Keep up the good work!

**Comment by Grabstertv** [comment]
+1

Baffles me that this still isn't a thing in an app that has "notes" in its name. Every note-taking app I've used has WYSIWYG markdown but an app that is supposed to be "simple" doesn't have it. It needs it. I like simplenote but going back and forth from preview and writing is not intuitive. Even Obsidian implemented live preview recently and it makes it so much easier to write. No live preview just makes writing a complete mess when you have a bunch of information right next to each other.

**Comment by dokker** [comment]
Yes please. It would make Simplenote to another level of usability.

**Comment by manueljenni** [comment]
I just downloaded Simplenote 10 minutes ago and this is so annoying - why doesn't it have this feature? Probably gonna uninstall because of this...

**Comment by EzequielColungaAcha** [comment]
As @manueljenni says, this is a go or no go feature.. Uninstalling right now.. I will have to keep using VS Code for .md 🤷🏻‍♀️

**Comment by rollecode** [comment]
<s>After 16 years of Simplenote I'm finally making the move and importing all my notes to [Standard Notes](https://standardnotes.com/). :+1: 

Simplenote is still good "raw cloud notepad", but for formatting I suggest to look elsewhere.

Edit: Scratch that, too much organizing to do and no backup options... I'm back using Simplenote.</s>

Edit: Scratch that, I've been using Obsidian since 2024.

**Comment by jeffreylsoffer** [comment]
I really wish **particuallary for code blocks** that Simplenote styled similarly to the preview.

![Image](https://github.com/user-attachments/assets/05f6bcc5-3e5b-4dbe-a0e4-fa831c46c732)

This is how [FSNotes](https://github.com/glushchenko/fsnotes) shows code blocks and spans and I think this is a nice approach that stills shows all the characters of the original markdown.

Also, a related question: why does Simplenote use a san-serif font for the headers but a serif font for the body but only for the markdown preview? It looks really odd.

![Image](https://github.com/user-attachments/assets/5e9208b5-bbb6-469c-b000-15cd7f21b87c)
